### PR TITLE
drpc: return right error when drpc delete fails

### PR DESCRIPTION
**Pull request by raghavendra-talur**
There was a bug in the code where if processDeletion() had an error and setProgressionAndUpdate() didn't have one, then we returned a nil error.

A test is added that checks for the required error.

### CODE REVIEW

**Review by ShyamsundarR — COMMENTED** [review]
LGTM, the envtests are failing though, PTAL, Thanks!

**Review by rakeshgm — APPROVED** [review]
I looked at the controller code. Looks good to me.

**Comment by rakeshgm** [comment]
@raghavendra-talur failure in `E2E/e2e-rdr (drenv)` is because of the CEL changes and all the fields in DRPolicy are now marked as required. the fix is merged in the main branch now. you may need to rebase your branch

File: controllers/drplacementcontrol_controller.go [hunk 1]
```diff
@@ -722,10 +722,14 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 		// then the DRPC should be deleted as well. The least we should do here is to clean up DPRC.
 		err := r.processDeletion(ctx, drpc, placementObj, logger)
 		if err != nil {
-			// update drpc progression only on err
 			logger.Info(fmt.Sprintf("Error in deleting DRPC: (%v)", err))
 
-			return ctrl.Result{}, r.setProgressionAndUpdate(ctx, drpc, rmn.ProgressionDeleting)
+			statusErr := r.setProgressionAndUpdate(ctx, drpc, rmn.ProgressionDeleting)
```
Comment (BenamarMk) [inline controllers/drplacementcontrol_controller.go:727]: This change is fine, however, given that we already log in line 725, we can just change the return to be:
```
return ctrl.Result{Requeue: true}, r.setProgressionAndUpdate(ctx, drpc, rmn.ProgressionDeleting)
```
But, we may want to keep it consistent and return an error when an error occurs. My attempt was to save the extra 5 lines of code.

**Comment by raghavendra-talur** [comment]
@BenamarMk @ShyamsundarR can we merge this?

**Review by ShyamsundarR — APPROVED** [review]

**Pull request merged**